### PR TITLE
errors: map locally-generated gRPC errors

### PR DESCRIPTION
**Pull request by goaway**
Description: Because Envoy may generate local grpc replies for grpc requests, this is required (for now) to maintain existing error mapping behavior.
Risk Level: Low
Testing: Pending

Signed-off-by: Mike Schore <mike.schore@gmail.com>

### CODE REVIEW

**Comment by goaway** [comment]
Planning to add integration coverage to this as a follow-up, similar to the existing HTTP coverage.

File: library/common/extensions/filters/http/local_error/filter.cc [hunk 1]
```diff
@@ -15,25 +17,75 @@ namespace Extensions {
...
-  // Absence of EnvoyUpstreamServiceTime header implies this is a local reply, which we treat as
-  // a stream error.
-  if (!success && headers.get(Http::Headers::get().EnvoyUpstreamServiceTime).empty()) {
+  const auto grpc_status = Grpc::Common::getGrpcStatus(headers);
+  if (grpc_status && grpc_status.value() != Grpc::Status::WellKnownGrpcStatus::Ok) {
+    ASSERT(end_stream,
+           "Local gRPC responses must consist of a single headers frame. If Envoy changes "
+           "this expectation, this code needs to be updated.");
+    ENVOY_LOG(debug, "intercepted local GRPC response");
+    grpcError_ = true;
```
Comment (rebello95) [inline library/common/extensions/filters/http/local_error/filter.cc:76]: It looks like this is never read, is it needed?

**Review by rebello95 — APPROVED** [review]
Looks good, pending tests in a future PR

File: library/common/extensions/filters/http/local_error/filter.cc (hunk 1 above)
Comment (goaway) [reply library/common/extensions/filters/http/local_error/filter.cc:76]: Actually you're right, it's not; good catch. I was using it previously, but was able to simplify things. I'll go ahead and remove.

**Review by rebello95 — APPROVED** [review]

**Pull request merged**